dashboard/app/controllers/api/v1/pd/workshop_attendance_controller.rb

```diff
@@ -44,7 +44,9 @@ def update
       no_longer_attending = existing_user_ids - attending_user_ids
 
       new_attendees.each do |user_id|
-        Pd::Attendance.create session: session, teacher: User.find_by_id!(user_id)
+        Retryable.retryable(tries: 2, on: ActiveRecord::RecordNotUnique) do
+          Pd::Attendance.find_or_create_by! session: session, teacher: User.find_by_id!(user_id)
+        end
```

Comment:
I didn't want the change of infinite loop (however unlikely), but I didn't realize we had that pattern already. I can remove the `tries` if you prefer.